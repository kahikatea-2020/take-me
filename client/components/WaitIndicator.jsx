import React from 'react'
import { connect } from 'react-redux'

const WaitIndicator = (props) => {
  return props.pending
    ? <img data-testid='wait-indicator' src='https://upload.wikimedia.org/wikipedia/commons/c/c7/Loading_2.gif' alt='loading indicator' style={{ width: '100px', height: '100px' }} />
    : null
}

function mapStateToProps (state) {
  return {
    pending: state.pending
  }
}

export default connect(mapStateToProps)(WaitIndicator)
