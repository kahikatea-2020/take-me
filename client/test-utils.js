import React from 'react'
import { Provider } from 'react-redux'
import { createStore, applyMiddleware, compose } from 'redux'
import { MemoryRouter as Router } from 'react-router-dom'
import thunkMiddleware from 'redux-thunk'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import reducer from './reducers'
const enhancer = compose(applyMiddleware(thunkMiddleware))

export const renderWithRedux = (
  ui,
  { initialState, initialEntries = ['/'], store = createStore(reducer, initialState, enhancer) } = {}
) => {
  return {
    ...render(
      <Provider store={store}>
        <Router initialEntries={initialEntries} initialIndex={0}>
          {ui}
        </Router>
      </Provider>
    ),
    store
  }
}
