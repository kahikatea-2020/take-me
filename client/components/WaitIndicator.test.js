import React from 'react'
import { renderWithRedux } from '../test-utils'

import WaitIndicator from './WaitIndicator'

test('Test the WaitIndicator', () => {
  const { getByTestId } = renderWithRedux(<WaitIndicator />, {
    initialState: { pending: true }
  })

  expect(getByTestId('wait-indicator')).toMatchSnapshot()
})
