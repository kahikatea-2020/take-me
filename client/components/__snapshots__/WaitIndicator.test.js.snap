// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the WaitIndicator 1`] = `
<img
  alt="loading indicator"
  data-testid="wait-indicator"
  src="/loadIndicator.gif"
  style="width: 50px;"
/>
`;
